d3servercheck.ps1

```diff
@@ -1,70 +1,52 @@
-function Write-Server-Result($tcpconns) {
+function Write-ServerResult($tcpconns) {
     if (!$tcpconns) {
         return
     }
-
-    foreach ($tcpconn in $tcpconns) {
-        Write-Host "$($tcpconn.RemoteAddress) (" -NoNewline
-        if ($is_good_ip.ContainsKey($tcpconn.RemoteAddress)) {
-            if ($is_good_ip[$tcpconn.RemoteAddress]) {
-                Write-Host "GOOD" -NoNewline
+    $Unique_Addresses = $tcpconns | Sort-object -property RemoteAddress -unique
+    write-host "Connections:"
+    foreach ($address in $Unique_Addresses) {
+        if ($is_good_ip.ContainsKey($address.RemoteAddress)) {
+            if ($is_good_ip[$address.RemoteAddress]) {
+                $Result = "GOOD"
             } else {
-                Write-Host "BAD" -NoNewline
+                $Result =  "BAD" 
             }
-        } else {
-            Write-Host "UNKNOWN" -NoNewline
         }
-
-        Write-Host ") " -NoNewline
+        else{
+            $Result = 'Unknown'
+        }
+        
+        $Message = '{0} : ({1})' -f $address.remoteaddress,$Result
+        Write-Host "$message"
     }
+    write-host "`r"
 }
 
 Write-Host "Starting d3servercheck"
 
 $csv = Import-Csv -Path servers.csv
 
+try {
+    $Diablo3_Process = get-process -name 'Diablo III64' -ErrorAction Stop
+}
+Catch{
+    write-error "Diablo 3 is not running. Exiting script..."
+    exit 1
+}
+
 $is_good_ip = @{}
 
 foreach ($ip in $csv) {
     $is_good_ip.Add($ip.server, $ip.status -eq "good")
 }
 
-$blizzard_server_ips_32 = @()
-$blizzard_server_ips_33 = @()
-$known_ips = $csv.server
-for ($j=0; $j -lt 256; $j++) {
-    $ip_32 = "37.244.32.$($j)"
-    $ip_33 = "37.244.33.$($j)"
-    if (!$is_good_ip.ContainsKey($ip_32)) {
-        $blizzard_server_ips_32 += $ip_32
-    }
-
-    if (!$is_good_ip.ContainsKey($ip_33)) {
-        $blizzard_server_ips_33 += $ip_33
-    }
-   
-}
-
 Write-Host "Read servers"
 Write-Host "Starting checks"
 
 while ($true) {
     Start-Sleep -s 5
 
-    $tcpconns_0 = Get-NetTCPConnection -RemoteAddress $known_ips -EA SilentlyContinue
-    $tcpconns_1 = Get-NetTCPConnection -RemoteAddress $blizzard_server_ips_32 -EA SilentlyContinue
-    $tcpconns_2 = Get-NetTCPConnection -RemoteAddress $blizzard_server_ips_33 -EA SilentlyContinue
-
-    if (!$tcpconns_0 -and !$tcpconns_1 -and !$tcpconns_2) {
-        Write-Host "no connection to Blizzard game servers detected"
-        continue;
-    }
-
-    Write-Host "connections: " -NoNewline
-
-    Write-Server-Result($tcpconns_0)
-    Write-Server-Result($tcpconns_1)
-    Write-Server-Result($tcpconns_2)
+    $Diablo3_ActiveConnections = get-NetTCPConnection -owningprocess $Diablo3_Process.id
```

Comment:
Fetching all of D3's connections leads to a lot of false positives, no? I saw several connections in the Resource Monitor when hacking this together, so I think that this might print a lot of IPs which are not _game server_ IPs, which are the only ones we're interested in. As it seems the EU gamer servers are located somewhere in `37.244.32.0/23`, so we should filter the IPs to this range, probably in the `Write-ServerResult` function.

Also inconsistent capitalization.